File: serve-spa.py
#!/usr/bin/env python3
"""
Simple SPA Server for Marden SEO React App
Handles client-side routing by serving index.html for all routes
"""

import os
import sys
from http.server import HTTPServer, SimpleHTTPRequestHandler
from pathlib import Path

class SPAHandler(SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory='dist-spa-temp', **kwargs)
    
    def do_GET(self):
        # Get the requested path
        path = self.translate_path(self.path)
        
        # If it's a file that exists, serve it normally
        if os.path.exists(path) and os.path.isfile(path):
            super().do_GET()
        # If it's a directory or non-existent path, serve index.html
        else:
            self.path = '/index.html'
            super().do_GET()

def run_server(port=8090):
    server_address = ('', port)
    httpd = HTTPServer(server_address, SPAHandler)
    
    print(f"🚀 Marden SEO React App running at http://localhost:{port}/")
    print("✅ SPA routing enabled - page refresh will work on all routes")
    print("Press Ctrl+C to stop the server")
    
    try:
        httpd.serve_forever()
    except KeyboardInterrupt:
        print("\n👋 Server stopped")
        sys.exit(0)

if __name__ == '__main__':
    run_server()

File: serve-spa.js
const express = require('express');
const path = require('path');
const fs = require('fs');

const app = express();
const PORT = 8090;
const DIST_DIR = path.join(__dirname, 'dist-spa-temp');

// Serve static files
app.use(express.static(DIST_DIR));

// Handle SPA routing - serve index.html for all routes
app.get('*', (req, res) => {
  const indexPath = path.join(DIST_DIR, 'index.html');
  
  // Check if the requested file exists (for assets, images, etc.)
  const requestedPath = path.join(DIST_DIR, req.path);
  
  if (fs.existsSync(requestedPath) && fs.statSync(requestedPath).isFile()) {
    // If it's a real file, serve it
    res.sendFile(requestedPath);
  } else {
    // Otherwise, serve index.html for React Router to handle
    res.sendFile(indexPath);
  }
});

app.listen(PORT, () => {
  console.log(`🚀 Marden SEO React App running at http://localhost:${PORT}/`);
  console.log('✅ SPA routing enabled - page refresh will work on all routes');
});
